esp-hal/src/timer/systimer.rs

```diff
@@ -723,33 +728,64 @@ pub mod etm {
     //!    The system timer can generate the following ETM events:
     //!    - SYSTIMER_EVT_CNT_CMPx: Indicates the alarm pulses generated by
     //!      COMPx
-    // FIXME(mabez)
+    //! ## Example
+    //! ```rust, no_run
+    #![doc = crate::before_snippet!()]
+    //! # use esp_hal::timer::systimer::{etm::Event, SystemTimer};
+    //! # use esp_hal::timer::PeriodicTimer;
+    //! # use esp_hal::etm::Etm;
+    //! # use esp_hal::gpio::{
+    //! #     etm::{Channels, OutputConfig},
+    //! #     Level,
+    //! #     Pull,
+    //! # };
+    //! # use fugit::ExtU32;
+    //! let syst = SystemTimer::new(peripherals.SYSTIMER);
+    //! let etm = Etm::new(peripherals.SOC_ETM);
+    //! let gpio_ext = Channels::new(peripherals.GPIO_SD);
+    //! let alarm0 = syst.alarm0;
+    //! let mut led = peripherals.GPIO1;
+    //!
+    //! let timer_event = Event::new(&alarm0);
+    //! let led_task = gpio_ext.channel0_task.toggle(
+    //!     &mut led,
+    //!     OutputConfig {
+    //!         open_drain: false,
+    //!         pull: Pull::None,
+    //!         initial_state: Level::High,
+    //!     },
+    //! );
+    //!
+    //! let _configured_etm_channel = etm.channel0.setup(&timer_event,
+    //! &led_task);
+    //!
+    //! let timer = PeriodicTimer::new(alarm0);
+    //! // configure the timer as usual
+    //! // when it fires it will toggle the GPIO
+    //! # }
+    //! ```
 
     use super::*;
 
     /// An ETM controlled SYSTIMER event
-    pub struct Event<'a> {
-        alarm: &'a mut Alarm,
+    pub struct Event {
+        id: u8,
```

Comment:
I wonder if s.th. like dropping the alarm could cause any harm. I guess not and holding `&'a mut Alarm` was over cautious (and annoying)

Reply:
If the alarm is dropped and the timer is stopped, the `Event` won't occur but I think this behavior is quite obvious. The `Event` itself is essentially inert but to create it it we need temporary access to the alarm to generate the correct id.

Reply:
basically what I thought - I don't think we disable alarms when they get dropped but that's not in scope of this PR

Reply:
Yeah we don't yet, but we probably will in the future